### PR TITLE
Add extract method support for ref struct interfaces

### PR DESCRIPTION
**Pull request by ToddGrun**
The method extraction code uses ITypeParameterSymbol's constraints to generate the new method's text. This codepath had not yet been modified to support ITypeParameterSymbol.AllowsRefLikeType

This is in support of the "allows ref struct" on interfaces feature outlined here: https://github.com/dotnet/roslyn/issues/72124

This ref structs for interfaces feature was merged via this PR: https://github.com/dotnet/roslyn/pull/73567

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/CodeGenerationSymbolFactory.cs [hunk 1]
```diff
@@ -337,10 +338,11 @@
         bool hasReferenceConstraint = false,
         bool hasUnmanagedConstraint = false,
         bool hasValueConstraint = false,
+        bool allowsRefLikeType = false,
```
Comment (AlekseyTs) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/CodeGenerationSymbolFactory.cs:341]: >bool allowsRefLikeType = false, [](http://example.com/codeflow?start=8&length=31)

Similar comment about order of parameters

Comment (AlekseyTs) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/CodeGenerationSymbolFactory.cs:341]: Especially that in `CodeGenerationTypeParameterSymbol` it comes before `hasUnmanagedConstraint` parameter.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/Symbols/CodeGenerationTypeParameterSymbol.cs [hunk 2]
```diff
@@ -18,6 +18,7 @@
     bool hasConstructorConstraint,
     bool hasReferenceConstraint,
     bool hasValueConstraint,
+    bool allowsRefLikeType,
```
[the remark this answers is not in the document]
Comment (ToddGrun) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/Symbols/CodeGenerationTypeParameterSymbol.cs]: re-ordered for the new parameter to be the last bool param

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/CodeGenerationSymbolFactory.cs [hunk 3]
```diff
@@ -322,7 +322,8 @@ public static ITypeParameterSymbol CreateTypeParameterSymbol(string name, int or
             attributes: default, varianceKind: VarianceKind.None,
             name: name, constraintTypes: [],
             hasConstructorConstraint: false, hasReferenceConstraint: false, hasValueConstraint: false,
-            hasUnmanagedConstraint: false, hasNotNullConstraint: false, ordinal: ordinal);
+            allowsRefLikeType: false, hasUnmanagedConstraint: false, hasNotNullConstraint: false,
```
Comment (ToddGrun) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/CodeGenerationSymbolFactory.cs:325]: >allowsRefLikeType: false,  [](http://example.com/codeflow?start=12&length=26)

missed this one, fixing

**Pull request merged**